src/ngraph/op/embedding_segments_sum.cpp

```diff
@@ -0,0 +1,210 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/embedding_segments_sum.hpp"
+#include "ngraph/op/constant.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::v3::EmbeddingSegmentsSum::type_info;
+
+op::v3::EmbeddingSegmentsSum::EmbeddingSegmentsSum(const Output<Node>& emb_table,
+                                                   const Output<Node>& indices,
+                                                   const Output<Node>& segment_ids,
+                                                   const Output<Node>& num_segments,
+                                                   const Output<Node>& per_sample_weights,
+                                                   const Output<Node>& default_index)
+    : Op({emb_table, indices, segment_ids, num_segments, per_sample_weights, default_index})
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v3::EmbeddingSegmentsSum::EmbeddingSegmentsSum(const Output<Node>& emb_table,
+                                                   const Output<Node>& indices,
+                                                   const Output<Node>& segment_ids,
+                                                   const Output<Node>& num_segments,
+                                                   const Output<Node>& per_sample_weights)
+    : Op({emb_table, indices, segment_ids, num_segments, per_sample_weights})
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v3::EmbeddingSegmentsSum::EmbeddingSegmentsSum(const Output<Node>& emb_table,
+                                                   const Output<Node>& indices,
+                                                   const Output<Node>& segment_ids,
+                                                   const Output<Node>& num_segments)
+    : Op({emb_table, indices, segment_ids, num_segments})
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v3::EmbeddingSegmentsSum::validate_and_infer_types()
+{
+    enum
```

Comment:
Let's make this an enum class and stick it in the class definition. Validation isn't the only thing that needs to look at inputs.

Reply:
that enum is just to make the validation routine more readable..dont see a reason why we need a enum class for it.

Reply:
Passes can use them too.